redbot/cogs/warnings/warnings.py

```
@@ -301,19 +348,37 @@ def __init__(self, bot: Red):
 
         await warning_points_add_check(self.config, ctx, user, current_point_count)
         try:
+            dm = await self.config.guild(ctx.guild).toggle_dm()
+            if dm:
+                em = discord.Embed(
+                    title=_("Warning from {user}").format(user=ctx.author),
+                    description=reason_type["description"],
+                )
+                em.add_field(name=_("Points"), value=str(reason_type["points"]))
+                await user.send(
+                    _("You have received a warning in {guild_name}.").format(
+                        guild_name=ctx.guild.name
+                    ),
+                    embed=em,
+                )
+        except discord.HTTPException:
```

Since the purpose of this is to suppress the exception, I recommend `contextlib.suppress()` and making sure the only thing inside the suppress block is the `await user.send(...)` part.